jwt.js

```diff
@@ -1,61 +1,114 @@
-'use strict'
-
 const fp = require('fastify-plugin')
 const JWT = require('jsonwebtoken')
-const assert = require('assert')
+const steed = require('steed')()
+
+function wrapStaticSecretInCallback (secret) {
+  return function (_, __, cb) {
+    return cb(null, secret)
+  }
+}
 
-function fastifyJWT (fastify, opts, next) {
-  if (!opts.secret) {
+function fastifyJwt (fastify, options, next) {
+  if (!options.secret) {
     return next(new Error('missing secret'))
   }
 
-  const secret = opts.secret
+  let secretCallback = options.secret
+
+  if (typeof secretCallback !== 'function') { secretCallback = wrapStaticSecretInCallback(secretCallback) }
+
+  let _requestProperty = options.userProperty || options.requestProperty || 'user'
+  let _resultProperty = options.resultProperty
```

Comment:
I would remove those. Why are they needed? I would not overwrite anything.

Reply:
See my more in depth answer below. TL;DR this is my first middleware and I abstracted this from `express-jwt`. If this is bad practice then I'll remove it.